Source/WebCore/rendering/RenderElement.h

```diff
@@ -195,7 +195,7 @@ class RenderElement : public RenderObject {
 
     // absoluteAnchorRectWithScrollMargin() is similar to absoluteAnchorRect, but it also takes into account any
     // CSS scroll-margin that is set in the style of this RenderElement.
-    LayoutRect absoluteAnchorRectWithScrollMargin(bool* insideFixed = nullptr) const;
+    std::pair<LayoutRect, LayoutRect> absoluteAnchorRectWithScrollMargin(bool* insideFixed = nullptr) const;
```

Comment:
Returning a std::pair here makes the code hard to read; you have to dig around to find out what `.first` and `.second` represent. It would be better to return a struct, either with two rects, or a rect and a margin.

Reply:
Fixed.